concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import hashlib
+import os
+import sys
+import time
+from datetime import datetime
+from threading import Thread
+
+import requests
+from golem_messages.shortcuts import dump
+
+from api_testing_common import api_request
+from api_testing_common import assert_condition
+from api_testing_common import count_fails
+from api_testing_common import get_force_get_task_result
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import run_tests
+from common.helpers import get_current_utc_timestamp
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+now = datetime.now()
+testing_subtask_id = str(now.year) + str(now.month) + str(now.day) + str(now.hour) + str(now.minute) + str(
+    now.second) + str(now.microsecond)
+
+responses = []  # type: str
+
+
+def multi_threads(func):
+    def wrapper(*args, **kwargs):
+        for i in range(3):
+            t = Thread(target=func, args=(*args,), kwargs={**kwargs})
+            t.start()
+
+    return wrapper
+
+
+@count_fails
+def test_case_that_multiple_requests_by_one_subtask_will_not_be_cause_of_server_error(cluster_consts, cluster_url, test_id):
+
+    send_correct_request(cluster_consts=cluster_consts, cluster_url=cluster_url)
+
+    time.sleep(4)
```

Comment:
I would put it in a `while` loop until `len(responses) = expected`.

Reply:
done in for loop with `break` if length expected. If there will be some error instead of response test would be never end. But generally great idea with such loop, will be save and much faster :)

Reply:
But I don't think that `for` loop guarantees that. It still asserts that response will come in 0,5 second. What if after all 3 or 9 runs there will not be all responses ? It will just break anyway. My idea was to run loop until all responses are there for sure. Off course you can count error responses as responses too. Or if you expect errors that are not returning responses, you can count exceptions or set a global time limit for it.

Reply:
done as discussed. I only left time.sleep(0.1), without this in local tests everything was longer